### PR TITLE
Cairo Zero Parser

### PR DESCRIPTION
**Pull request by krafugo**
# Description

- Add Cairo Zero structs
- Add test cases to check marshaling

# Comment

- Field debug-info is still needed to be completed

### CODE REVIEW

File: pkg/parsers/zero/zero_test.go [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+			expected: ZeroProgram{
+				Hints: map[string][]Hint{
+					"0": {{
+						AccessibleScopes: []string{"starkware.cairo.common.math", "starkware.cairo.common.math.assert_not_equal"},
+						Code:             "from starkware.cairo.lang.vm.relocatable import RelocatableValue\nboth_ints = isinstance(ids.a, int) and isinstance(ids.b, int)\nboth_relocatable = (\n    isinstance(ids.a, RelocatableValue) and isinstance(ids.b, RelocatableValue) and\n    ids.a.segment_index == ids.b.segment_index)\nassert both_ints or both_relocatable, \\\n    f'assert_not_equal failed: non-comparable values: {ids.a}, {ids.b}.'\nassert (ids.a - ids.b) % PRIME != 0, f'assert_not_equal failed: {ids.a} = {ids.b}.'",
+					}},
+				},
+			},
+		},
+		// TODO: Add debug-info test case
```
Comment (rodrigo-pino) [inline pkg/parsers/zero/zero_test.go:91]: 👀 todo

File: pkg/parsers/zero/zero.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
+package zero
+
+import (
+	"encoding/json"
+	"os"
+)
+
+type ZeroProgram struct {
+	Attributes []string          `json:"attributes"`
+	Builtins   map[string]int64  `json:"builtins"`
```
Comment (rodrigo-pino) [inline pkg/parsers/zero/zero.go:10]: Could you use the Builtin type we defined for parsers/starknet. This way validation already occurs during parsing

**Review by rodrigo-pino — COMMENTED** [review]
Great work! Some comments here and there

File: pkg/parsers/zero/zero_test.go (hunk 1 above)
Comment (krafugo) [reply pkg/parsers/zero/zero_test.go:91]: This was just to point out the missing debug-info definition that you told me was not important at the moment. Should I remove this message?

File: pkg/parsers/zero/zero.go (hunk 2 above)
Comment (krafugo) [reply pkg/parsers/zero/zero.go:10]: I think they are not the same type. The starknet-parser one is uint8.

Comment (rodrigo-pino) [reply pkg/parsers/zero/zero.go:10]: Yes, but it has it's own `Builtin` type, with its implementation of a marsharller/unmarshaller. The same can be reused here if I am not mistaken.

File: pkg/parsers/zero/zero_test.go (hunk 1 above)
Comment (rodrigo-pino) [reply pkg/parsers/zero/zero_test.go:91]: Let's have a simple test only

**Review by rodrigo-pino — APPROVED** [review]
Structure matches the output of a Cairo Zero compiled program and there are unit tests for each field.

**Pull request merged**